test/integration/supervisor_tls_spec_test.go

```diff
@@ -0,0 +1,266 @@
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "invalid-oidc-idp-invalid-kind",
+			expectedError:      `The OIDCIdentityProvider "invalid-oidc-idp-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should disallow certificate authority data source with invalid kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: invalid-oidc-idp-invalid-kind
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: sorcery
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "invalid-oidc-idp-invalid-kind",
+			expectedError:      `The OIDCIdentityProvider "invalid-oidc-idp-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "sorcery": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should create a custom resource passing all validations using a Secret source",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: valid-oidc-idp-secret-kind1
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: Secret
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "valid-oidc-idp-secret-kind1",
+			expectedError:      "",
+		},
+		{
+			name: "should create a custom resource passing all validations using a ConfigMap source",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: valid-oidc-idp-cm-kind1
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: ConfigMap
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`)),
+			customResourceName: "valid-oidc-idp-cm-kind1",
+			expectedError:      "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resourceName := tc.customResourceName
+			expectedError := tc.expectedError
+			yamlBytes := tc.customResourceYaml
+			t.Parallel()
```

Comment:
Usually `t.Parallel()` is the first line of a test, just to make it stand out more and to be consistent.

Reply:
Calling `t.Parallel` is not liked by the [loopclosure analyzer.](https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/loopclosure). In our case, bc we are using go > 1.22 it should be ok to remove these local variables making `t.Parallel` the first thing in the function